### ISSUE TITLE
MixedIntegerLinearProgram.show: Factor out methods show_constraints etc.

### ISSUE THREAD

**Issue by mkoeppe** [sagemath/sage#34885]
<div id="comment:0"></div>

Also improve formatting of equations

CC:  @mantepse

Component: **linear programming**

_Issue created by migration from https://trac.sagemath.org/ticket/34885_

**Comment by alexandremr01** [comment]
What do you think would be important for improving equations? 

I was considering and thought about something like this for the architecture:

I see that `MixedIntegerLinearProgram.show` only uses the `MixedIntegerLinearProgram` object to access the `_backend` and `linear_functions_parent`. This second one only depends on the base ring, that is also in the backend. Thus, formatting only depends on the backend. We could otherwise have

- A n abstract class `Formatter` class with methods `show`, `show_variables`, `show_constraints`, `show_objective`.
- Something like `PlainTextFormatter`, implementing current behavior
- Something like `LatexFormatter`, implementing LaTex format
- The `show` method of `MixedIntegerLinearProgram` would accept an argument `format` and select the proper Formatter.

Would this be interesting?

**Comment by mkoeppe** [comment]
Yes, better formatting for MIPs would be a very welcome contribution.
Note that Sage has existing infrastructure for latexing. Classes are expected to define a method `_latex_`; the global function `latex` dispatches to it. So this would be the preferred interface. Also see the IPython magic `%display latex`.

In addition to the `MixedIntegerLinearProgram` class, Sage also has a didactical implementation of LP called "InteractiveLP", which has good support for latexing. Some code can probably be reused/refactored.

**Comment by mkoeppe** [comment]
> What do you think would be important for improving equations?

Currently equations are shown as ranged constraints; that should definitely be improved.
